flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/metrics/MetricStore.java

```diff
@@ -570,6 +578,19 @@ void retainSubtasks(Set<Integer> activeSubtasks) {
             subtasks.keySet().retainAll(activeSubtasks);
         }
 
+        void removeTransientMetrics(int subtaskIndex) {
+            if (subtasks.containsKey(subtaskIndex)) {
+                // Remove in both places as task metrics are duplicated in task metric store and
+                // subtask metric store for metrics query on WebInterface.
+                metrics.keySet()
```

Comment:
I prefer not specifically mentioning 'query on WebInterface' since the metric store also exposes metrics to users through the REST API.

Reply:
Addressed, thx